include/dxc/Support/HLSLVersion.h

```diff
@@ -0,0 +1,37 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// HLSLOptions.h                                                             //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Support for command-line-style option parsing.                            //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef LLVM_HLSL_HLSLVERSION_H
+#define LLVM_HLSL_HLSLVERSION_H
+
+namespace hlsl {
+
+ // This Updates to this enum must be reflected in HLSLOptions.td and Options.td
+ // for the hlsl_version option.
+enum class LangStd : unsigned long  {
+  vUnset = 0,
+  vError = 1,
+  v2015 = 2015,
+  v2016 = 2016,
+  v2017 = 2017,
+  v2018 = 2018,
+  v2021 = 2021,
+  v202x = 2029,
```

Comment:
My thought process was that we'll probably do an update before the end of the decade, so I set the max for only the last digit. One of the benefits of this being an enum is that we won't actually allow implicit direct comparison or conversion from int, so hopefully the actual value doesn't really matter here.

I had considered instead making the macro `vNext`, but it occurred to me that we might at some point in the future have more than one "future" language in flight at the same time. Clang had this problem and ended up using C++1x, C++1y and C++1z to differentiate 11, 14, and 17.

Very open to other thoughts. I mostly was adding this option so that we had a place to put Union support since I don't think it will make HLSL 2021.